forge-game/src/main/java/forge/game/ability/effects/ChangeZoneEffect.java

```diff
@@ -1143,6 +1143,11 @@ else if (origin.contains(ZoneType.Hand) && player.isOpponentOf(decider)) {
                             fetchList = CardLists.filter(fetchList, Predicates.not(CardPredicates.sharesCMCWith(c)));
                         }
                     }
+                    if (sa.hasParam("DifferentPower")) {
+                        for (Card c : chosenCards) {
+                            fetchList = CardLists.filter(fetchList, Predicates.not(CardPredicates.sharesPowerWith(c)));
```

Comment:
Personally I would avoid the overhead in the other files and just use this line:
`fetchList = CardLists.filter(fetchList, Predicates.not(Predicates.compose(Predicates.equalTo(c.getNetPower()), CardPredicates.Accessors.fnGetNetPower)));`